### ISSUE TITLE
testRelease should fail if making docs fails

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#6239]
It seems that ``util/buildRelease/testRelease`` does not fail if its attempt to make module documentation fails, and this can spiral into other errors downstream.  Specifically, if the temporary modules/standard/SysCTypes.chpl used to generate documentation is not removed (as it is not if this build step fails), the compiler will rightfully complain of an ambiguous module.  It seems to me that a failure to make the docs should kill the testRelease script at that point rather than leaving open the possibility of rippling into other errors or having the error go undetected.  

That said, I'm not sure offhand what it would take to do this.  From what I can tell, 'make module-docs' returns an error code (2), so it would seem to be something upstream from that step.  The following shows a case that went wrong when the deinit() procedure in ChapelLocale.chpl was not marked as "no doc" (and removing the "no doc" pragma should permit this to be reproduced, at least until such time as that internal error is fixed).  If testRelease worked as I believe it should, the following lines should've been the last ones seen in the log rather than lines much further on when trying to do 'make test' on the Chapel compiler itself.  This would make the source of the problem easier to debug and also reduce the time to failure if bad code is checked in.

````
Generating module docs from $CHPL_HOME/modules/ into rst/modules
(cd ../modules && make documentation)
mkdir -p standard
cd standard && /private/var/folders/69/jbk0y4q95lq8xhhwpqgwf_rh000bxs/T/chapel-release.bradc.deleteme.hlmkk/chapel-test/util/config/make_sys_basic_types.py --doc SysCTypes.chpl
Generating module documentation with chpldoc
export CHPLDOC_AUTHOR='Cray Inc' && \
	/private/var/folders/69/jbk0y4q95lq8xhhwpqgwf_rh000bxs/T/chapel-release.bradc.deleteme.hlmkk/chapel-test/bin/darwin/chpldoc --save-sphinx /private/var/folders/69/jbk0y4q95lq8xhhwpqgwf_rh000bxs/T/chapel-release.bradc.deleteme.hlmkk/chapel-test/modules/sphinx --no-html standard/Assert.chpl standard/Barrier.chpl standard/BigInteger.chpl standard/BitOps.chpl standard/Buffers.chpl standard/CommDiagnostics.chpl standard/DateTime.chpl standard/DynamicIters.chpl standard/FileSystem.chpl standard/GMP.chpl standard/Help.chpl standard/IO.chpl standard/List.chpl standard/Math.chpl standard/Memory.chpl standard/Path.chpl standard/Random.chpl standard/Reflection.chpl standard/Regexp.chpl standard/Spawn.chpl standard/Sys.chpl standard/SysBasic.chpl standard/SysError.chpl standard/Time.chpl standard/Types.chpl standard/UtilReplicatedVar.chpl standard/SysCTypes.chpl dists/BlockCycDist.chpl dists/BlockDist.chpl dists/CyclicDist.chpl dists/DimensionalDist2D.chpl dists/PrivateDist.chpl dists/ReplicatedDist.chpl dists/StencilDist.chpl dists/dims/BlockCycDim.chpl dists/dims/BlockDim.chpl dists/dims/ReplicatedDim.chpl layouts/LayoutCSR.chpl  packages/BLAS.chpl packages/Curl.chpl packages/FFTW.chpl packages/FFTW_MT.chpl packages/Futures.chpl packages/HDFS.chpl packages/HDFSiterator.chpl packages/LAPACK.chpl packages/LinearAlgebra.chpl packages/MPI.chpl packages/Norm.chpl packages/OwnedObject.chpl packages/RangeChunk.chpl packages/RecordParser.chpl packages/Search.chpl packages/SharedObject.chpl packages/Sort.chpl packages/VisualDebug.chpl packages/ZMQ.chpl internal/Atomics.chpl internal/ChapelArray.chpl internal/ChapelComplex_forDocs.chpl internal/ChapelIO.chpl internal/ChapelIteratorSupport.chpl internal/ChapelLocale.chpl internal/UtilMisc_forDocs.chpl internal/ChapelRange.chpl internal/String.chpl internal/ChapelSyncvar.chpl internal/ChapelTuple.chpl internal/ChapelEnv.chpl
internal error: assertion error [AstToText.cpp:185]
make[3]: *** [documentation] Error 1
make[2]: *** [module-docs] Error 2
````

**Comment by bradcray** [comment]
@awallace-cray and/or @ben-albrecht : Could you look into this?

**Comment by ben-albrecht** [comment]
At a glance, it looks like this block in `testReleaseHelp` was intended to catch the error:

```perl

$mymake -j$num_procs
set tmpstatus = $status
if ($tmpstatus != 0) then
    echo "ERROR: make failed"
    exit($tmpstatus)
endif
rehash   # required for csh only; if we convert this to bash, drop this
set tmpstatus = $status
if ($tmpstatus != 0) then
    echo "ERROR: make failed"
    exit($tmpstatus)
endif

```

I'll try to reproduce the error, and see if I can find out why we don't catch the docs failure earlier.

**Comment by ben-albrecht** [comment]
The problem is caused by the `doc/Makefile` user-friendly error message consuming the exit code (`$ERRCODE`):

```Makefile

docs: FORCE
	@../util/run-in-venv.bash $(MAKE) html-release || (echo;echo "If a documentation hierarchy change was made on the master branch, \
	try resolving with: \`git clean "'$$CHPL_HOME'"/doc/*\`";echo)
```

meaning a failed `make docs` still returns exit code 0.

I see 2 viable options:

1) Drop this user-friendly error message, since we don't expect documentation hierarchy to change too much in the future (... right?)

2) Emit error messages, such that the exit code is preserved.

**Comment by ben-albrecht** [comment]
Resolved in #6256
